gdx/src/com/badlogic/gdx/math/noise/FastNoiseLite.java

```diff
@@ -0,0 +1,2375 @@
+		-0.6553931414f, 0.7381471625f, 0.159995615f, 0, 0.3910961323f, 0.9188871375f, -0.05186755998f, 0, -0.4879022471f,
+		-0.5904376907f, 0.6429111375f, 0, 0.6014790094f, 0.7707441366f, -0.2101820095f, 0, -0.5677173047f, 0.7511360995f,
+		0.3368851762f, 0, 0.7858573506f, 0.226674665f, 0.5753666838f, 0, -0.4520345543f, -0.604222686f, -0.6561857263f, 0,
+		0.002272116345f, 0.4132844051f, -0.9105991643f, 0, -0.5815751419f, -0.5162925989f, 0.6286591339f, 0, -0.03703704785f,
+		0.8273785755f, 0.5604221175f, 0, -0.5119692504f, 0.7953543429f, -0.3244980058f, 0, -0.2682417366f, -0.9572290247f,
+		-0.1084387619f, 0, -0.2322482736f, -0.9679131102f, -0.09594243324f, 0, 0.3554328906f, -0.8881505545f, 0.2913006227f, 0,
+		0.7346520519f, -0.4371373164f, 0.5188422971f, 0, 0.9985120116f, 0.04659011161f, -0.02833944577f, 0, -0.3727687496f,
+		-0.9082481361f, 0.1900757285f, 0, 0.91737377f, -0.3483642108f, 0.1925298489f, 0, 0.2714911074f, 0.4147529736f,
+		-0.8684886582f, 0, 0.5131763485f, -0.7116334161f, 0.4798207128f, 0, -0.8737353606f, 0.18886992f, -0.4482350644f, 0,
+		0.8460043821f, -0.3725217914f, 0.3814499973f, 0, 0.8978727456f, -0.1780209141f, -0.4026575304f, 0, 0.2178065647f,
+		-0.9698322841f, -0.1094789531f, 0, -0.1518031304f, -0.7788918132f, -0.6085091231f, 0, -0.2600384876f, -0.4755398075f,
+		-0.8403819825f, 0, 0.572313509f, -0.7474340931f, -0.3373418503f, 0, -0.7174141009f, 0.1699017182f, -0.6756111411f, 0,
+		-0.684180784f, 0.02145707593f, -0.7289967412f, 0, -0.2007447902f, 0.06555605789f, -0.9774476623f, 0, -0.1148803697f,
+		-0.8044887315f, 0.5827524187f, 0, -0.7870349638f, 0.03447489231f, 0.6159443543f, 0, -0.2015596421f, 0.6859872284f,
+		0.6991389226f, 0, -0.08581082512f, -0.10920836f, -0.9903080513f, 0, 0.5532693395f, 0.7325250401f, -0.396610771f, 0,
+		-0.1842489331f, -0.9777375055f, -0.1004076743f, 0, 0.0775473789f, -0.9111505856f, 0.4047110257f, 0, 0.1399838409f,
+		0.7601631212f, -0.6344734459f, 0, 0.4484419361f, -0.845289248f, 0.2904925424f, 0};
+
+	// This section defines the math functions used by FastNoiseLite.
+	// Note: The prefix "fast" is irrelevant to its actual speed/performance.
+
+	private static float fastMin (final float a, final float b) {
+		return Math.min(a, b);
+	}
+
+	private static float fastMax (final float a, float b) {
+		return Math.max(a, b);
+	}
+
+	private static float fastAbs (final float f) {
+		return f < 0 ? -f : f;
+	}
+
+	private static float fastSqrt (final float f) {
+		return (float)Math.sqrt(f);
+	}
+
+	private static int fastFloor (final float f) {
+		return f >= 0 ? (int)f : (int)f - 1;
+	}
+
+	private static int fastRound (final float f) {
+		return f >= 0 ? (int)(f + 0.5f) : (int)(f - 0.5f);
+	}
+
+	private static float lerp (final float a, final float b, final float t) {
+		return a + t * (b - a);
+	}
+
+	private static float interpHermite (final float t) {
+		return t * t * (3 - 2 * t);
+	}
+
+	private static float interpQuintic (final float t) {
+		return t * t * t * (t * (t * 6 - 15) + 10);
```

Comment:
This interpolation can exceed the 0 to 1 range when passed certain less-than-1 floats. The lowest input that goes out-of-range is `interpQuintic(0.99535805f) == 1.0000001f`, which is surprisingly far from 1.0f. `interpQuintic(243f/244f) == 1.0000002f` is also true. I use `return t * t * t * (t * (t * 6f - 15f) + 9.999998f);` , myself, which won't exceed the boundary.